pkg/scheduler/objects/application.go

```diff
@@ -198,9 +200,18 @@ func NewApplication(siApp *si.AddApplicationRequest, ugi security.UserGroup, eve
 	app.rmEventHandler = eventHandler
 	app.rmID = rmID
 	app.appEvents = newApplicationEvents(app, events.GetEventSystem())
+	app.appEvents.sendNewApplicationEvent()
 	return app
 }
 
+func (sa *Application) SetEnableAppStateEvent(enable bool) {
+	sa.enableAppStateEvent = enable
+}
+
+func (sa *Application) ResetAppEvents() {
+	sa.appEvents = newApplicationEvents(sa, events.GetEventSystem())
+}
```

Comment:
Do not export these two methods. In fact, these are only called from unit tests, so I suggest that you move them, to `application_test.go` (see examples like `addPlaceholderDataWithLocking()`, `handleApplicationEventWithLocking()`). Obviously `enableAppStateEvent` must stay as a member variable.

Reply:
Makes sense. Thanks.